code/game/machinery/vitals_monitor.dm

```diff
@@ -50,76 +72,157 @@
 				breathing = "normal"
 			else if(lungs.breath_fail_ratio < 1)
 				breathing = "shallow"
-		
+
 		to_chat(user, SPAN_NOTICE("Breathing: [breathing]"))
+	if(connected_optable)
+		to_chat(user, SPAN_NOTICE("Connected to adjacent [connected_optable]."))
 
 /obj/machinery/vitals_monitor/Process()
 	if(QDELETED(victim))
-		victim = null
+		update_victim()
 	if(victim && !Adjacent(victim))
-		victim = null
-		update_use_power(POWER_USE_IDLE)
+		update_victim()
+	if(connected_optable && !Adjacent(connected_optable))
+		update_victim()
+		update_optable()
 	if(victim)
 		update_icon()
-	if(beep && victim && victim.pulse())
-		playsound(src, 'sound/machines/quiet_beep.ogg')
-	
+
+/obj/machinery/vitals_monitor/proc/update_victim(var/new_victim = null)
+	var/old_victim = victim
+	victim = new_victim
+	if(victim)
+		visible_message(SPAN_NOTICE("\The [src] is now showing data for \the [victim]."))
+	else
+		if(old_victim != new_victim) // Protects against qdel edge case. In all other cases we want a message printed.
+			visible_message(SPAN_NOTICE("\The [src] is no longer showing data from [isnull(old_victim)? "any patient" : "\the [old_victim]"]."))
+	update_use_power(isnull(victim)? POWER_USE_IDLE : POWER_USE_ACTIVE)
+	update_icon()
+
+/obj/machinery/vitals_monitor/proc/update_optable(obj/machinery/optable/new_optable = null)
+	if(new_optable == connected_optable)
+		return
+	if(connected_optable) //gotta clear existing connections first
+		connected_optable.connected_monitor = null
+	connected_optable = new_optable
+	if(connected_optable)
+		connected_optable.connected_monitor = src
+		visible_message(SPAN_NOTICE("\The [src] is now relaying information from \the [connected_optable]"))
+		//In case there's already a patient on the table
+		update_victim(connected_optable.victim)
+	else
+		visible_message(SPAN_NOTICE("\The [src] is no longer relaying data from a connected operating table."))
+
 /obj/machinery/vitals_monitor/MouseDrop(over_object, src_location, over_location)
 	if(!CanMouseDrop(over_object))
 		return
+	update_optable()
 	if(victim)
-		victim = null
-		update_use_power(POWER_USE_IDLE)
+		update_victim()
 	else if(ishuman(over_object))
-		victim = over_object
-		update_use_power(POWER_USE_ACTIVE)
-		visible_message(SPAN_NOTICE("\The [src] is now showing data for [victim]."))
+		update_victim(over_object)
+	else if(istype(over_object, /obj/machinery/optable/))
+		var/obj/machinery/optable/new_table_connection = over_object
+		update_optable(new_table_connection)
 
 /obj/machinery/vitals_monitor/on_update_icon()
 	overlays.Cut()
```

Comment:
While you're modifying the file it looks like it would be trivial to change this to use `cut_overlays()` and `add_overlay()`.

Reply:
👍